grunt/service-worker.js

```diff
@@ -35,6 +35,7 @@ function writeServiceWorkerFile({staticDirectoryPath, apiSrcDirectoryPath, scrip
     dynamicUrlToDependencies: {
       '/': [path.join(staticDirectoryPath, 'templates', 'inbox.html')],
       '/medic/login': [path.join(apiSrcDirectoryPath, 'templates/login', 'index.html')],
+      '/medic/_design/medic/_rewrite/': [path.join(apiSrcDirectoryPath, 'public', 'appcache-upgrade.html')],
```

Comment:
This won't work if your DB is not named "medic" which is a supported use case. I can't think of any way to solve this without modifying the service worker at runtime which would be a shame. Possibly we can ignore this as an edge case of an edge case (db not named medic, user offline when upgrading). What do you think?

Reply:
Didn't we "alias" all DB names to be "medic"? So even if when I'm using some other db name, the url pattern will still be `/medic/_design/medic/_rewrite`? 

https://github.com/medic/medic/blob/master/api/src/routing.js#L191

Reply:
I don't think aliasing it in API would solve the issue, because we need the request to be caught by the service worker when they're offline, right? So if the 3.4 URL is `/totally-not-medic/_design/medic/_rewrite` then when they install the SW, disconnect, and restart we need to catch the request at the old URL to do the redirect. Or have I misunderstood something?

`app_settings` and `update_settings` and the line you cite all use the `environment.db` setting which is set from the `COUCH_URL` environment variable so it'll work no matter what the db name is set to. As far as I can tell we don't hardcode the db name anywhere, and tests should be failing if we do. If you can find anywhere where the db name is hardcoded it's a bug and please raise an issue for it!

Reply:
I've created https://github.com/medic/medic/issues/5692 to illustrate what I'm talking about.

> If you can find anywhere where the db name is hardcoded it's a bug and please raise an issue for it!

Can you clarify the intent of the alias created [here](https://github.com/medic/medic/blob/master/api/src/routing.js#L191)? My assumption was that all user facing URLs should appear with `medic` as the dbname but internally API will use `not_medic`. If this is true, it should be fine to hardcode the dbname as 'medic` in >v3.5. But maybe I've misunderstood?

You're definitely right about the upgrade path being broken when `dbname != medic`. Pushing a change for that now so upgrades are at possible, but this "edge case" (User will get error page when opening the app as an offline user + dbname not medic).

Reply:
Yes that's right in 3.5+ - because one port has one API which has one DB (no matter what it's called) we can assume every request to that port is for that DB. My concern was about 3.4 upgrades which request the not-medic URL and aren't online so don't get the redirect/proxy.